src/arch/riscv/rvdbg013.c

```diff
@@ -1582,8 +1587,13 @@ static bool decode_load_store_inst(target *t, uint32_t dpc, target_addr *watch)
 
 static int riscv_breakwatch_set(target *t, struct breakwatch *bw)
 {
-	uint32_t mcontrol = CSR_MCONTROL_DMODE | CSR_MCONTROL_ACTION_DEBUG |
-		CSR_MCONTROL_ENABLE_MASK;
+	int ret = 0;
+	bool free_trigger = false;
+	uint32_t tselect_saved, tdata1;
+	uint32_t trigger_idx = 0U;
+	uint16_t trigger_info = 0U;
+	uint32_t mcontrol = CSR_TDATA1_TYPE(2) | CSR_MCONTROL_DMODE |
```

Comment:
Please check for whitespace here too or run "git rebase --whitespace=fix origin/ruabmbua"

Reply:
Here, CSR_MCONTROL_ACTION_DEBUG is shifted with spaces (not tabs) in order to make it aligned with the upper CSR_TDATA1_TYPE(2) in _every_ editor, whatever the displayed width of the tab indentation by users. This is not an indentation actually.

Reply:
git log --check flags this red:
"src/arch/riscv/rvdbg013.c:1596: indent with spaces.
+                           CSR_MCONTROL_ACTION_DEBUG | CSR_MCONTROL_ENABLE_MASK;
"
Use as much tabs as possible and only fill the remainder with space.

Reply:
I didn't know of --check option, nice. But reading its help, it says a whitespace error is "a space character that is immediately followed by a tab character inside the initial indent of the line". Here, it is the opposite: tabs first for indentation, followed by a few space chars to align with upper line, as shown in this screenshot (tabs and spaces displayed with arrows and dots):
![image](https://user-images.githubusercontent.com/3934692/150778574-4f8cc426-3720-41b5-8d3c-fc2bdae44f07.png)

Also, git log --check doesn't report any error on my side:
```shell
$ git blame -L1502,+1 -- src/arch/riscv/rvdbg013.c
ffc9982b (Fabrice Prost-Boucle 2022-01-18 22:33:01 +0100 1502) ...
$ git log -n 1 --check ffc9982b
commit ffc9982b577121e91078c74b4f8cf3b38dee7c0e
Author: Fabrice Prost-Boucle <fabalthazar@falbalab.fr>
Date:   Tue Jan 18 22:33:01 2022 +0100

    Set triggers of type 2 explicitly and which have type 2 available.
    This supports triggers that do not have 'type' preset to 2.
```
git version 2.34.1

False positive?

Reply:
If you open the file in doubt in the review aspect on this page, you see solid darker green blocks for spaces where tab could be used. Just for your info, I hope not to annoy you

Reply:
Apologies, my answer referred to line 1502 instead of 1596.
But I still see none "indent with spaces" around 1596.
Can you see this screenshot?
![image](https://user-images.githubusercontent.com/3934692/150826815-b9a836dc-2bf5-40ab-b03c-238a2d6bd0c3.png)
The only spaces you see are before CSR_MCONTROL_ACTION_DEBUG, to align with above CSR_TDATA1_TYPE(2). It will stay aligned whatever editor is used and how width it displays tabs.
These spaces follow one single tab which is a real indentation level.

In other words, there are a lot of spaces here but seem legitimate as long as the indentation level is respected.

I feel I start to bother you about a point I cannot get. grrr :)